### PR TITLE
8295430: Use cmsDoTransformLineStride instead of cmsDoTransform in the loop

### PR DESCRIPTION
**Pull request by mrserb**
This is the request to improve the change made by the
[8005530: [lcms] Improve performance of ColorConverOp for default destinations](http://hg.openjdk.java.net/jdk8/jdk8/jdk/rev/fd8810d50c99)

Right now the LCMSTransform mantain the special "imageAtOnce" flags and call cmsDoTransform function in the loop per line to support the gaps betwen the lines in the images . This code can be improved by using one call to cmsDoTransformLineStride. Some discussion about this method can be found [here](https://github.com/mm2/Little-CMS/issues/314), the full [specification](https://github.com/mm2/Little-CMS/blob/master/doc/LittleCMS2.14%20API.pdf):

```
void cmsDoTransformLineStride(cmsHTRANSFORM Transform,
                              const void* InputBuffer,
                              void* OutputBuffer,
                              cmsUInt32Number PixelsPerLine,
                              cmsUInt32Number LineCount,
                              cmsUInt32Number BytesPerLineIn,
                              cmsUInt32Number BytesPerLineOut,
                              cmsUInt32Number BytesPerPlaneIn,
                              cmsUInt32Number BytesPerPlaneOut);

This function translates bitmaps with complex organization. Each bitmap may contain
several lines, and every may have padding. The distance from one line to the next one is
BytesPerLine{In/Out}. In planar formats, each line may hold several planes, each plane may
have padding. Padding of lines and planes should be same across all bitmap. I.e. all lines
in same bitmap have to be padded in same way. This function may be more efficient that
repeated calls to cmsDoTransform(), especially when customized plug-ins are being used.

Parameters:

-  hTransform: Handle to a color transform object.
-  InputBuffer: A pointer to the input bitmap
-  OutputBuffer: A pointer to the output bitmap.
-  PixelsPerLine: The number of pixels for line, which is same on input and in output.
-  LineCount: The number of lines, which is same on input and output
-  BytesPerLine{In,Out}: The distance in bytes from one line to the next one.
-  BytesPerPlaneIn{In,Out}: The distance in bytes from one plane to the next one inside a line. Only applies
                            in planar formats. 


Notes:
- This function is quite efficient, and is used by some plug-ins to give a big
  speedup. If you can load whole image to memory and then call this function
  over it, it will be much faster than any other approach.
- BytesPerPlaneIn{In,Out} is ignored if the formats used are not planar. Please note
  1-plane planar is indistinguishable from 1-component chunky, so BytesPerPlane is
  ignored as well in this case.
- If the image does not have any gaps between the pixels and lines BytesPerLine{}
  are equal to the pixel's size * PixelsPerLine.
- To specify padding between pixels, use T_EXTRA() and EXTRA_SH() to specify
  extra channels.
```

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8295430](https://bugs.openjdk.org/browse/JDK-8295430): Use cmsDoTransformLineStride instead of cmsDoTransform in the loop

### Reviewers
 * [Phil Race](https://openjdk.org/census#prr) (@prrace - **Reviewer**) ⚠️ Review applies to [1c017aac](https://git.openjdk.org/jdk/pull/10754/files/1c017aacb21337e30044251a030c37bd7c3f1a3e)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10754/head:pull/10754` \
`$ git checkout pull/10754`

Update a local copy of the PR: \
`$ git checkout pull/10754` \
`$ git pull https://git.openjdk.org/jdk pull/10754/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10754`

View PR using the GUI difftool: \
`$ git pr show -t 10754`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10754.diff">https://git.openjdk.org/jdk/pull/10754.diff</a>

</details>

### CODE REVIEW

File: src/java.desktop/share/native/liblcms/LCMS.c [hunk 1]
```diff
@@ -548,18 +547,11 @@ JNIEXPORT void JNICALL Java_sun_java2d_cmm_lcms_LCMS_colorConvert
...
-        cmsDoTransform(sTrans, inputRow, outputRow, width * height);
-    } else {
-        for (int i = 0; i < height; i++) {
-            cmsDoTransform(sTrans, inputRow, outputRow, width);
-            inputRow += srcNextRowOffset;
-            outputRow += dstNextRowOffset;
-        }
-    }
+    cmsDoTransformLineStride(sTrans, input, output, width, height,
+                             srcNextRowOffset, dstNextRowOffset, 0, 0);
```
Comment (mrserb) [inline src/java.desktop/share/native/liblcms/LCMS.c:554]: We do not use "planar" types, so the last two parameters are ignored.

**Comment by prrace** [comment]
This all looks good and I see that cmsDoTransform ends up in the same place as cmsDoTransformLineStride
so the performance of the "simple" case should be no worse.

I am just waiting for my test run to complete before approving.

**Review by prrace — APPROVED** [review]
Tests all passed

**Pull request closed without merging**